node/src/actors/chain_manager/mining.rs

```diff
@@ -538,6 +561,100 @@ impl ChainManager {
         }
     }
 
+    /// Create a superblock and forward it
+    fn superblock_forwarding(
+        &mut self,
+        ctx: &mut Context<Self>,
+        current_epoch: u32,
+        superblock_period: u32,
+        ars_members: Vec<PublicKeyHash>,
+        genesis_hash: Hash,
+    ) {
+        let superblock_index = current_epoch / superblock_period;
+
+        let inventory_manager = InventoryManager::from_registry();
+
+        let init_epoch = current_epoch - superblock_period;
+        let init_epoch = init_epoch.saturating_sub(1);
+        let final_epoch = current_epoch.saturating_sub(2);
+
+        futures::future::ok(self.handle(
+            GetBlocksEpochRange::new_with_limit(init_epoch..=final_epoch, 0),
+            ctx,
+        ))
+        .and_then(move |res| match res {
+            Ok(v) => {
+                let block_hashes: Vec<Hash> = v.into_iter().map(|(_epoch, hash)| hash).collect();
+                futures::future::ok(block_hashes)
+            }
+            Err(e) => {
+                log::error!("{}", e);
```

Comment:
Add some context to this errors?
```suggestion
                log::error!("Error in GetBlocksEpochRange: {}", e);
```